### PR TITLE
[infra] Fix npm install with version 9

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
### Context

After upgrading to `npm` version 9 I was no longer able to install the repo. This seems to be due to npm 9 being more strict about missing files.

### Fix

Fix was to remove a devDependency on a missing folder. I briefly considered adding every test package but they seem to work without being a dep, so opted to remove.

This should be a transparent change with no impact except for fixing npm 9 install.

### Repro

To repro, ensure you are on `npm` version 9. I was reproing with `9.1.1`. Run `npm install` on Lit.

Notice the following error:

```
npm ERR! enoent ENOENT: no such file or directory, open '/x/x/x/lit/packages/labs/analyzer/test-files/basic-elements/package.json'
```

### CODE REVIEW

**Review by augustjk — APPROVED** [review]
Removal makes sense to me since we don't reference `lit-analyzer-test-files` with the bare specifier anywhere. Thank you for finding and fixing this!

**Comment by AndrewJakubowicz** [comment]
Thank you @augustjk for confirming the issue and helping me debug it!

File: packages/labs/analyzer/package.json [hunk 1]
```diff
@@ -58,8 +58,5 @@
   "dependencies": {
     "package-json-type": "^1.0.3",
     "typescript": "~4.7.4"
-  },
-  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (AndrewJakubowicz) [reply packages/labs/analyzer/package.json:62]: This doesn't break on `npm` major version `8`. It's a silent failure.

[the remark this answers is not in the document]
Comment (kevinpschaaf) [reply packages/labs/analyzer/package.json:62]: Not sure why it was there. It came in your initial commit: https://github.com/lit/lit/commit/df10f6d34bd59b736cb723250e0b02b2cc5012e3#diff-c2362ee734f7c4b58aea6224689e537cf6e86bdcf155c22e752887481554b101R34

Seems fine to remove. In tests, we point the analyzer directly at the filesystem path containing the test files, so it doesn't need to be linked into node_modules.

**Pull request merged**